bin/chunk_clusters.py

```diff
@@ -8,81 +8,86 @@
     )
     parser.add_argument(
-        "-t",
-        "--threshold",
-        required=True,
-        metavar="INT",
-        type=int,
-        help="Minimum cluster size to keep.",
+        "-t", "--threshold", required=True, metavar="INT", type=int, help="Minimum cluster size to keep."
     )
     parser.add_argument(
-        "-o",
-        "--out_folder",
-        required=True,
-        metavar="FOLDER",
-        type=str,
-        help="Name of the output folder to be created.",
+        "-p", "--threads", metavar="INT", type=int, default=4, help="Number of threads for parallel fasta writing (default: 4)."
+    )
+    parser.add_argument(
+        "-o", "--out_folder", required=True, metavar="FOLDER", help="Name of the output folder to be created."
     )
     return parser.parse_args(args)
 
 
 def collect_clusters(clustering_file, threshold):
-    # Collect clusters with a size threshold, storing in a defaultdict
     clusters = defaultdict(list)
-
     with open(clustering_file) as f:
-        csv_reader = csv.reader(f, delimiter="\t")
-        for row in csv_reader:
+        reader = csv.reader(f, delimiter="\t")
+        for row in reader:
             rep, member = row
             clusters[rep].append(member)
-
-    # Filter clusters by threshold
     return {
         rep: members for rep, members in clusters.items() if len(members) >= threshold
     }
 
 
+def load_sequences(fasta_file, needed_ids):
+    sequences = {}
+    with open(fasta_file) as handle:
+        for record in SeqIO.parse(handle, "fasta"):
+            if record.id in needed_ids:
+                sequences[record.id] = record
+    return sequences
+
+
+def write_cluster(chunk_num, members, sequences, out_folder):
+    output_file = os.path.join(out_folder, f"{chunk_num}.fasta")
+    with open(output_file, "w") as out_handle:
+        for member in members:
+            if member in sequences:
+                SeqIO.write(sequences[member], out_handle, "fasta")
+    return output_file
+
+
 def main(args=None):
     args = parse_args(args)
 
-    # Create output directory if it doesn't exist
     os.makedirs(args.out_folder, exist_ok=True)
 
-    # Collect clusters that meet the threshold
-    clusters = collect_clusters(args.clustering, int(args.threshold))
+    # Step 1: Parse clusters
+    clusters = collect_clusters(args.clustering, args.threshold)
+    print(f"Clusters filtered.")
+
+    # Step 2: Collect sequence IDs only from clusters above threshold
+    needed_ids = set()
+    for members in clusters.values():
+        needed_ids.update(members)
+    print("Required sequence IDs collected.")
+
+    # Step 3: Load only sequences of clusters above threshold
+    sequences = load_sequences(args.sequences, needed_ids)
+    print("Required sequences loaded.")
 
-    # Use pyfastx for indexed FASTA access (no need to load sequences into memory)
-    fasta = pyfastx.Fasta(args.sequences)
```

Comment:
One thing here.. this is building a sqlite db as index. That could cause IO contention issues.